invenio_rdm_migrator/streams/runner.py

```diff
@@ -118,15 +102,8 @@ def run(self):
                 stream.run()
                 # sucessfully finished stream run, now we can dump that stream cache
                 for name, cache in self.cache.items():
-                    if (
-                        name == "communities" or name == "max_pid"
-                    ):  # FIXME: implement communities cache
-                        continue
                     cache_file = self.cache_dir / f"{name}.json"
                     cache.dump(cache_file)
             except Exception:
                 self.logger.error(f"Stream {stream.name} failed.", exc_info=1)
                 continue
-        # dump the max pid generated in pids cache
-        # call global pid_pk to update the cache with the max value
-        self._dump_max_pid_cache()
```

Comment:
IMO it is better if the cache is dumped as the other on a per successful stream basis. The reason is the "run without ET steps" capabilities that were merged in another PR:
- Before: if one stream failed the migration would need to be re-tried from scratch. Therefore it would make sense to keep the max pid pk to the initial value (i.e. not dump).
- Now: since n number of streams can be just "loaded" (i.e. only the SQL COPY command would be executed). The max pid pk needs to be dumped per stream to avoid collisions when running using existing data.